sklearn/utils/validation.py

```diff
@@ -1220,6 +1222,12 @@ def check_symmetric(array, *, tol=1e-10, raise_warning=True, raise_exception=Fal
         and array.transpose(). If sparse, then duplicate entries are first
         summed and zeros are eliminated.
     """
+    xp, array_api = get_namespace(array.data)
+
+    if array_api:
+        array.row = array.row._array
+        array.col = array.col._array
+        array.data = array.data._array
```

Comment:
I think the the pattern of unpacking and repacking the array goes against the principle of `array_api`. The rest of the implementation will end up doing all the work as on the CPU and then at the end convert it back to GPU.